nbdime/tests/test_server_extension.py

```diff
@@ -96,26 +96,44 @@ def test_diff_api(git_repo2, server_extension_app):
     assert data['diff']
     assert len(data.keys()) == 2
 
-@pytest.mark.timeout(timeout=WEB_TEST_TIMEOUT)
-def test_git_diff_api(git_repo2, server_extension_app):
+@pytest.mark.timeout(timeout=WEB_TEST_TIMEOUT*6)
+def test_git_diff_api(git_repo2, server_extension_app, filespath):
     local_path = os.path.relpath(git_repo2, server_extension_app['path'])
     url = 'http://127.0.0.1:%i/nbdime/api/gitdiff' % server_extension_app['port']
-    r = requests.post(
-        url, headers=auth_header,
-        data=json.dumps({
-            'ref_prev': {
-                'git': 'HEAD'
-            },
-            'ref_curr': {
-                'special': 'WORKING'
-            },
-            'file_path': pjoin(local_path, 'diff.ipynb')
-        }))
-    r.raise_for_status()
-    data = r.json()
-    nbformat.validate(data['base'])
-    assert data['diff']
-    assert len(data.keys()) == 2
+
+    # Add a difference betweeen index and working tree:
+    shutil.copy(
+        pjoin(filespath, 'foo--1.ipynb'),
+        pjoin(git_repo2, 'sub', 'subfile.ipynb')
+    )
+
+    def _make_ref(key):
+        if key.lower() in ('working', 'index'):
+            return {'special': key}
+        return {'git': key}
+
+    # Test various diffs:
+    for args in (
+        ('HEAD', 'WORKING', 'diff.ipynb'),
+        ('HEAD', 'INDEX', 'diff.ipynb'),
+        ('INDEX', 'HEAD', 'diff.ipynb'),
+        ('INDEX', 'WORKING', 'sub/subfile.ipynb'),
+        ('index', 'working', 'sub/subfile.ipynb'),
+        ('iNdeX', 'WorKING', 'sub/subfile.ipynb'),
+    ):
+        print(args)
+        r = requests.post(
+            url, headers=auth_header,
+            data=json.dumps({
+                'ref_local': _make_ref(args[0]),
```

Comment:
The reason I went with `prev` and `curr` was because a request like [1] to diff between the Working Tree and `origin/HEAD`, which can get a bit confusing since 'origin/HEAD' is a remote, and WORKING is actually the local changes. I'm fine with this naming if the intention is to keep things consistent.

[1] 

```
    "ref_local": {
        "git": "origin/HEAD"
        
    },
    "ref_remote": {
        "special": "WORKING"
    },
```